### PR TITLE
WIP: Avoid adding ellipsis if text does not overflow

### PR DESCRIPTION
**Pull request by bernardobelchior**
## Description

Avoid adding an ellipsis (`...`) if the text does not overflow. This should not only fix an issue, but also improve performance slightly as we don't need to check for overflow for tick labels that we know will fit. 

## Related Issue

https://github.com/recharts/recharts/issues/4735

## Motivation and Context

TBD

## How Has This Been Tested?

  indices: number[];
  type: 'set' | 'delete';

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

**Comment by ckifer** [comment]
maybe you and @razzeee should collaborate on this

https://github.com/recharts/recharts/pull/4740 

multiple things going on in `Text` at once

File: storybook/stories/API/component/Text.stories.tsx [hunk 1]
```diff
@@ -10,22 +10,27 @@ export default {
...
     lineHeight: '1em',
     maxLines: 3,
     scaleToFit: false,
     textAnchor: 'start',
-    verticalAnchor: 'end',
+    verticalAnchor: 'start',
     angle: 0,
-    width: 200,
-    content: 'This is really long text',
+    containerWidth: '100%',
```
Comment (ckifer) [inline storybook/stories/API/component/Text.stories.tsx:31]: I don't think this should work - Surface is just an svg which does not take relative width/height values

**Comment by bernardobelchior** [comment]
Closed in favor of https://github.com/recharts/recharts/pull/4743

**Comment by bernardobelchior** [comment]
> maybe you and @razzeee should collaborate on this
> 
> #4740
> 
> multiple things going on in `Text` at once

I think these changes are orthogonal: they won't interfere with each other, but once I'm done I'll see if there's anything we need to do 👍

**Pull request closed without merging**